src/server/auth/server/util.go

```diff
@@ -10,14 +10,15 @@ import (
 
 // CheckClusterIsAuthorizedInTransaction returns an error if the current user doesn't have
 // the permissions in `p` on the cluster
-func CheckClusterIsAuthorizedInTransaction(txnCtx *txnenv.TransactionContext, p ...auth.Permission) error {
-	me, err := txnCtx.Client.WhoAmI(txnCtx.ClientContext, &auth.WhoAmIRequest{})
+func (a *apiServer) CheckClusterIsAuthorizedInTransaction(txnCtx *txnenv.TransactionContext, p ...auth.Permission) error {
+	// TODO (actgardner): WhoAmIInTransaction
+	me, err := a.WhoAmI(txnCtx.ClientContext, &auth.WhoAmIRequest{})
```

Comment:
WhoAmI goes from an RPC to a method call, so it will benefit from the metadata being cached in the client context and avoid a postgres round-trip.